packages/cmrjs/index.js

```diff
@@ -20,58 +20,37 @@ const logDetails = {
 };
 
 /**
- * Search for a concept on CMR
  *
- * @param {string} type - the concept type. Choices are: collection, granule
- * @param {Object} searchParams - cmr search params
+ * @param {string} type - Concept type to search, choices: ['collections', 'granules']
+ * @param {Object} searchParams - CMR search parameters
  * @param {Array} existingResults - array of results returned in previous recursive calls
- * @returns {Promise.<Array>} an array of search results
+ * @returns {Array} - array of search results.
  */
-async function searchConcept(type, searchParams, existingResults) {
-  const limit = process.env.CMR_LIMIT || 100;
-  const pageSize = process.env.CMR_PAGE_SIZE || 50;
+async function searchConcept(type, searchParams, existingResults = []) {
```

Comment:
I've renamed some variables to make it a little easier to read, feel free to take it or ignore it:

```js
async function searchConcept(type, searchParams, previousResults = []) {
  const resultsLimit = process.env.CMR_LIMIT || 100;
  const pageSize = searchParams.pageSize || process.env.CMR_PAGE_SIZE || 50;

  const defaultParams = { page_size: pageSize };

  const url = `${getUrl('search')}${type}.json`;

  let pageNum;
  if (searchParams.page_num) pageNum = searchParams.page_num + 1;
  else pageNum = 1;

  const query = { ...defaultParams, ...searchParams, page_num: pageNum };

  const response = await got.get(url, { json: true, query });
  
  const fetchedResults = response.body.feed.entry || [];
  
  const results = [...previousResults, ...fetchedResults];

  const cmrHasMoreResults = results.length < response.headers['cmr-hits'];
  const resultsLimitReached = results.length >= resultsLimit;

  if (cmrHasMoreResults && !resultsLimitReached) {
    return searchConcept(type, query, results);
  }

  return results.slice(0, resultsLimit);
}
```